service/storage_service.cc

```diff
@@ -5408,11 +5410,12 @@ void storage_service::init_messaging_service(sharded<service::storage_proxy>& pr
                 curr_cdc_gen_id = ss._topology_state_machine._topology.current_cdc_generation_id;
             }
 
-            std::optional<canonical_mutation> cdc_generation_mutation;
+            std::vector<canonical_mutation> cdc_generation_mutations;
             if (curr_cdc_gen_id) {
                 // We don't need to fetch this data under group0 apply mutex, it's immutable.
-                // We only need to include the current CDC generation data in the snapshot,
-                // because nodes only load whatever `current_cdc_generation_id` points to in topology.
+                // We only need to include the current CDC generation data in the snapshot
+                // and the generation of data is being broadcasted but hasn't been entirely
+                // applied yet.
```

Comment:
I would get rid of this last sentence, it was useful to explain this when we were only pulling the current generation, now we're pulling everything so it's unnecessary.

Reply:
Fixed.